src/algebra/helpers.jl

```diff
@@ -5,4 +5,17 @@ import LinearAlgebra: Diagonal
 xT_A_x(x::AbstractVector, A::AbstractMatrix) = dot(x, A * x)
 
 diageye(::Type{T}, n::Int) where { T <: Real } = Matrix{T}(I, n, n)
-diageye(n::Int)                                = diageye(Float64, n)
\ No newline at end of file
+diageye(n::Int)                                = diageye(Float64, n)
+
+function softmax(x::Array{Float64,1})
```

Comment:
We can add `LogExpFunctions.jl` dependency with `softmax` and `sigmoid` functions included. Not sure about `dtanh`. Didn't find any equivalent in other libraries

Reply:
Changed the `sigmoid` function to the `logistic` function from `StatsFuns`. Created an alias for easier usage. Apparently we already had (and still have) a `softmax` function in `ReactiveMP` in `math.jl`. However, this one clamps its output for inference purposes, but I am not sure whether this is desirable. Perhaps we should rename that one to `softmax_clamped` to prevent issues if we want to use the unclamped version of the softmax function. Partly resolved by 5ed8820.